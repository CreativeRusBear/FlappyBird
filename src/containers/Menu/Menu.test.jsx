import React from "react";
import renderer from "react-test-renderer";
import {BrowserRouter} from "react-router-dom";

import {configure, shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import MenuWithRouter, {Menu} from "./Menu";

configure({adapter: new Adapter()});

describe('<Menu>', () => {
  const props = {
    history: {
      goBack: jest.fn(),
      location: {
        pathname: '/'
      }
    },
  }
  let wrapper;
  beforeEach(() => wrapper = shallow(<Menu {...props}/>));

  it('should renders correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <MenuWithRouter/>
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('testing Back <button>', () => {
    it('should hidden if open in the main page', () => expect(wrapper.find('button').prop('hidden')).toEqual(true));
    it('shouldn\'t be hidden if open the author page', () => {
      wrapper.setProps({
        history: {
          goBack: jest.fn(),
          location: {
            pathname: '/author'
          }
        }
      });
      expect(wrapper.find('button').prop('hidden')).toEqual(false);
    });
    it('should has event listener that running goBack prop function when click on button', () => {
      expect(wrapper.find('button').prop('onClick')()).toEqual(props.history.goBack());
    })
  })
});
