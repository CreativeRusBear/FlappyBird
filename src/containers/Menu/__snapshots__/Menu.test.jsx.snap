// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu> should renders correctly 1`] = `
<div
  className="Menu"
>
  <button
    hidden={true}
    onClick={[Function]}
  >
    Back
  </button>
  <div
    className="List"
  >
    <button>
      <a
        aria-current={null}
        href="/game"
        onClick={[Function]}
      >
        Play
      </a>
    </button>
    <button>
      <a
        aria-current={null}
        href="/author"
        onClick={[Function]}
      >
        Author
      </a>
    </button>
  </div>
  <div
    className="Links"
  >
    <a
      className="License"
      href="https://github.com/corocoto/flappy-bird/blob/master/LICENSE"
      rel="noopener noreferrer license"
      target="_blank"
      title="Link on the licence"
    >
       
    </a>
    <a
      className="Repo"
      href="https://github.com/corocoto/flappy-bird"
      rel="noopener noreferrer bookmark"
      target="_blank"
      title="Link on the repository"
    >
       
    </a>
  </div>
</div>
`;
